File: hfopenmusic/src/main/java/com/hfopenmusic/sdk/listener/NoDoubleClickListener.java
package com.hfopenmusic.sdk.listener;

import android.view.View;

public abstract class NoDoubleClickListener implements View.OnClickListener {


    public static final int MIN_CLICK_DELAY_TIME = 500;//这里设置不能超过多长时间
    private long lastClickTime = 0;

    protected abstract void onNoDoubleClick(View v);
    @Override
    public void onClick(View v) {
        long currentTime = System.currentTimeMillis();
        if (currentTime - lastClickTime > MIN_CLICK_DELAY_TIME) {
            lastClickTime = currentTime;
            onNoDoubleClick(v);
        }
    }
}

File: hfplayer/src/main/java/com/hf/player/utils/NoDoubleClickListener.kt
package com.hf.player.utils

import android.view.View

abstract class NoDoubleClickListener : View.OnClickListener {
    private var lastClickTime: Long = 0
    protected abstract fun onNoDoubleClick(v: View?)
    override fun onClick(v: View) {
        val currentTime = System.currentTimeMillis()
        if (currentTime - lastClickTime > MIN_CLICK_DELAY_TIME) {
            lastClickTime = currentTime
            onNoDoubleClick(v)
        }
    }

    companion object {
        const val MIN_CLICK_DELAY_TIME = 1000 //这里设置不能超过多长时间
    }
}